### PR TITLE
Elemental Analysis - Linking

### LINKED ISSUE #23426: Elemental Analysis - Linking Subsections
Should pull together the separate parts of the Elemental Analysis GUI.

### PR DESCRIPTION
**Pull request by ewancook**
**Description of work.**
Pulls together the Elemental Analysis MVP Modules in (an admittedly) messy `Elemental_Analysis.py` - this will be addressed in a future maintenance issue (#23498).

**Report to:** Aidy

**To test:**
The GUI should function as expected

Notes:
- Lines will update from the element selector (right click) when closed (not immediately when clicked)
- Lines will cycle colours - unfortunately, there's not much I can do about this right now

Closes #23426

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Elemental_Analysis.py [hunk 1]
```diff
@@ -39,46 +41,166 @@ def __init__(self, parent=None):
...
+        self.gamma_lines = []
+        self.plotting.update_canvas()
+
+    def _get_electron_peaks(self):
+        return self.ptable.peak_data["Electrons"].copy()
+
+    def _plot_electrons(self, subplot, colour=None):
+        if colour is None:
+            colour = self.line_colours.next()
+        for peak, intensity in iteritems(self.electron_peaks):
```
Comment (AnthonyLim23) [inline scripts/Elemental_Analysis.py]: `intensity` is not use, add a comment as to why it is here

File: scripts/Elemental_Analysis.py [hunk 2]
```diff
@@ -39,46 +41,166 @@ def __init__(self, parent=None):
...
+                subplot.axvline(
+                    float(peak), 0, 1, color=colour))
+        self.plotting.update_canvas()
+
+    def electrons_checked(self):
+        colour = self.line_colours.next()
+        for subplot_name, subplot in iteritems(self.plotting.get_subplots()):
+            self._plot_electrons(subplot, colour=colour)
+
+    def electrons_unchecked(self):
```
Comment (AnthonyLim23) [inline scripts/Elemental_Analysis.py]: can merge with gamma lines, do in issue ....

File: scripts/Elemental_Analysis.py [hunk 3]
```diff
@@ -89,51 +211,126 @@ def _generate_element_data(self):
...
+            self.load_run(detector, last_run)
+        colour = self.line_colours.next()
+        for element in self.ptable.selection:
+            for label, x_value in iteritems(self.element_data[element.symbol]):
+                l = self.plotting.get_subplot(detector).axvline(
+                    x_value, 1, 0, color=colour)
+                try:
+                    self.element_lines[element.symbol][x_value].append(l)
+                except KeyError:
+                    self.element_lines[element.symbol][x_value] = [l]
```
[the remark this answers is not in the document]
Comment (AnthonyLim23) [reply scripts/Elemental_Analysis.py]: line 214

**Comment by AnthonyLim23** [comment]
need to make sure processes end when GUI is closed

**Review by AnthonyLim23 — APPROVED** [review]
looks good

**Comment by soininen** [comment]
Please remember to note the person in the 'Report to' field that this has been merged.

**Pull request merged**